### PR TITLE
BaSP-TG-18: Time-sheets GET & POST tests

### PR DESCRIPTION
**Pull request by fede230298**
GET Tests:

- Status should be 200 if the body is empty
- Status should be 200, correct query, empty result (Here I want to ask if the status should be 404 instead of 200 because there is time sheet found

POST Tests:

- Status should be 201, should create a new time-sheet
- Status should be 400 if the body is empty
- Status should be 400 when the description is shorter than 3 characters
- Status should be 400 when the description is larger than 300 characters
- Status should be 400 when the date is not ISO formated
- Status should be 400 when the hours are negative
- Status should be 400 when the task is shorter than 24 characters
- Status should be 400 when the task is larger than 24 characters
- Status should be 400 when the employee is shorter than 24 characters
- Status should be 400 when the employee is larger than 24 characters
- Status should be 400 when the project is shorter than 24 characters
- Status should be 400 when the project is larger than 24 characters

### CODE REVIEW

**Comment by fede230298** [comment]
Please, do not take into account the last commit, will be fixed shortly

File: src/seed/employees.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import mongoose from 'mongoose';
+
+export default [
+  {
+    _id: mongoose.Types.ObjectId('634c68f3658f142935ea7f6e'),
```
Comment (lmalvasia) [inline src/seed/employees.js:5]: We should have almost 5 elements inside the seeds.

File: src/seed/projects.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import mongoose from 'mongoose';
+
+export default [{
+  _id: mongoose.Types.ObjectId('6352dae0282ae0599b7acd92'),
```
Comment (lmalvasia) [inline src/seed/projects.js:4]: We should have almost 5 elements inside the seeds

File: src/seed/tasks.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import mongoose from 'mongoose';
+
+export default [{
+  _id: mongoose.Types.ObjectId('6352dae0282ae0599b7acd91'),
```
Comment (lmalvasia) [inline src/seed/tasks.js:4]: We should have almost 5 elements inside the seeds

File: src/test/time-sheets.test.js [hunk 4]
```diff
@@ -1,3 +1,157 @@
...
+
+  describe('GET /time-sheets', () => {
+    test('status should be 200', async () => {
+      const response = await request(app).get('/time-sheets').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.data.length).toBeGreaterThan(0);
+    });
+    test('status should be 200, correct query, empty result', async () => {
```
Comment (lmalvasia) [inline src/test/time-sheets.test.js:41]: Great!!!!

File: src/test/time-sheets.test.js [hunk 5]
```diff
@@ -1,3 +1,157 @@
...
+  });
+
+  describe('GET /time-sheets', () => {
+    test('status should be 200', async () => {
+      const response = await request(app).get('/time-sheets').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.data.length).toBeGreaterThan(0);
+    });
```
Comment (lmalvasia) [inline src/test/time-sheets.test.js:40]: Add a new line here to separate the tests

File: src/test/time-sheets.test.js [hunk 6]
```diff
@@ -1,3 +1,157 @@
...
+  });
+
+  describe('POST /time-sheets', () => {
+    test('status should be 201, should create a new time-sheet', async () => {
+      const response = await request(app).post('/time-sheets').send(mockedTimeSheet);
+      expect(response.status).toBe(201);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+    });
```
Comment (lmalvasia) [inline src/test/time-sheets.test.js:56]: Add a new line here to separate the tests (do it in every test 😂 )

File: src/seed/employees.js (hunk 1 above)
Comment (fede230298) [reply src/seed/employees.js]: Seeds added!

File: src/seed/projects.js (hunk 2 above)
Comment (fede230298) [reply src/seed/projects.js:4]: Seeds added!

File: src/seed/tasks.js (hunk 3 above)
Comment (fede230298) [reply src/seed/tasks.js:4]: Seeds added!

File: src/test/time-sheets.test.js (hunk 5 above)
Comment (fede230298) [reply src/test/time-sheets.test.js:40]: New line added!

File: src/test/time-sheets.test.js (hunk 6 above)
Comment (fede230298) [reply src/test/time-sheets.test.js:57]: New line added among all tests! 😂

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**